web/app/scripts/black-spots/black-spots-directive.js

```diff
@@ -1,20 +1,90 @@
-(function () {
+(function() {
     'use strict';
 
     /* ngInject */
-    function BlackSpots() {
+    function BlackSpots($q, BoundaryState, LeafletDefaults, InitialState, FilterState,
+        RecordState, TileUrlService, QueryBuilder, BlackspotSets) {
+        var cartoDBAttribution = '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>';
+        var defaultLayerOptions = {
+            attribution: 'PRS',
+            detectRetina: true
+        };
+
         var module = {
-            restrict: 'EA',
-            templateUrl: 'scripts/black-spots/black-spots-partial.html',
-            bindToController: true,
+            restrict: 'A',
+            scope: false,
             replace: true,
-            controller: 'BlackSpotsController',
-            controllerAs: 'blackspots'
+            controller: '',
+            require: ['leafletMap'],
+            link: link
         };
         return module;
+
+        function link(scope, element, attrs, controllers) {
+            InitialState.ready().then(function() {
+                var leafletController = controllers[0];
+
+                leafletController.getMap().then(addBaseLayers);
+                var maxDateString;
+                if (FilterState.filters.hasOwnProperty('__dateRange')) {
```

Comment:
If `__dateRange` doesn't exist in the filters, the query should still be sent, but with the current datetime as `effective_at`. If the user hasn't manually edited the date range filter yet (e.g. the first time they load the page), this variable won't be set, but we should still show the current black spots.